File: src/java/CdbWebPortal/src/java/gov/anl/aps/cdb/common/exceptions/InvalidRequest.java
/*
 * Copyright (c) 2014-2015, Argonne National Laboratory.
 *
 * SVN Information:
 *   $HeadURL: $
 *   $Date: $
 *   $Revision: $
 *   $Author: $
 */
package gov.anl.aps.cdb.common.exceptions;

import gov.anl.aps.cdb.common.constants.CdbStatus;

/**
 * Invalid request exception.
 */
public class InvalidRequest extends CdbException {

    /**
     * Default constructor.
     */
    public InvalidRequest() {
        super();
        setErrorCode(CdbStatus.CDB_INVALID_REQUEST);
    }

    /**
     * Constructor using error message.
     *
     * @param message error message
     */
    public InvalidRequest(String message) {
        super(message);
        setErrorCode(CdbStatus.CDB_INVALID_REQUEST);
    }

    /**
     * Constructor using throwable object.
     *
     * @param throwable throwable object
     */
    public InvalidRequest(Throwable throwable) {
        super(throwable);
        setErrorCode(CdbStatus.CDB_INVALID_REQUEST);
    }

    /**
     * Constructor using error message and throwable object.
     *
     * @param message error message
     * @param throwable throwable object
     */
    public InvalidRequest(String message, Throwable throwable) {
        super(message, throwable);
        setErrorCode(CdbStatus.CDB_INVALID_REQUEST);
    }

}

File: src/python/cdb/common/exceptions/invalidRequest.py
#!/usr/bin/env python

"""
Copyright (c) UChicago Argonne, LLC. All rights reserved.
See LICENSE file.
"""


#
# Invalid request error class.
#

#######################################################################

from cdb.common.constants import cdbStatus 
from cdb.common.exceptions.cdbException import CdbException

#######################################################################

class InvalidRequest(CdbException):
    def __init__ (self, error='', **kwargs):
        CdbException.__init__(self, error, cdbStatus.CDB_INVALID_REQUEST, **kwargs)
